tests/sqlparser_mssql.rs

```diff
@@ -324,6 +324,26 @@ fn parse_delimited_identifiers() {
     //TODO verified_stmt(r#"UPDATE foo SET "bar" = 5"#);
 }
 
+#[test]
+fn parse_table_name_in_square_brackets() {
+    let select = ms().verified_only_select(r#"SELECT [a column] FROM [a schema].[a table]"#);
+    if let TableFactor::Table { name, .. } = only(select.from).relation {
+        assert_eq!(
+            vec![
+                Ident::with_quote('[', "a schema"),
```

Comment:
that is definitely not what I would have expected (I guess I would have expected a single '[a schema]' or '['. 'a schema, ']'

Reply:
I wanted to implement the feature, so I started by adding a test, and realized it was already there :smile: SQLPage used to parse everything with the generic parser, so mssql identifiers in square brackets didn't work, but they do work in the mssql dialect.

The current behavior seems to match how microsoft sql server parses the queries itself, so it's alright.

Reply:
Working on this project has illustrated to me so many peculiarities of SQL dialects. I feel sometimes like I am becoming a connoisseur 🤪